docs/en/testing/supplemental.md

```diff
@@ -158,6 +123,187 @@ describe('MyWidget', () => {
+	});
+});
+```
+
+The harness automatically mocks a number of core middlewares that will be injected into any middleware that requires them:
+
+-   invalidator
+-   setProperty
+-   destroy
+
+#### Dojo Mock Middleware
+
+There are a number of mock middlewares available to support testing widgets that use Dojo middleware. The mocks export a factory used to create the scoped mock middleware to be used in each test.
+
+##### Mock `node` Middleware
+
+Using `createMockNodeMiddleware` from `@dojo/framework/testing/mocks/middleware/node` creates a mock node middleware. To set the expected return from the node mock, call the `mockNode` with a `key` and expected node.
+
+##### Mock `intersection` Middleware
+
+Using `createMockIntersectionMiddleware` from `@dojo/framework/testing/mocks/middleware/intersection` creates a mock intersection middleware. To set the expected return from the intersection mock, call the `mockIntersection` with a `key` and expected intersection details.
+
+##### Mock `resize` Middleware
+
+Using `createMockResizeMiddleware` from `@dojo/framework/testing/mocks/middleware/resize` creates a mock resize middleware. To set the expected return from the resize mock, call the `mockResize` with a `key` and expected content rects.
+
+```tsx
+import { create, tsx } from '@dojo/framework/core/vdom'
+import resize from '@dojo/framework/core/middleware/resize'
+
+const factory = create({ resize });
+
+export const MyWidget = factory(function MyWidget({ middleware }) => {
+	const  { resize } = middleware;
+	const contentRects = resize.get('root');
+	return <div key="root">{JSON.stringify(contentRects)}</div>;
+});
+```
+
+```tsx
+import { tsx } from '@dojo/framework/core/vdom';
+import createMockResizeMiddleware from '@dojo/framework/testing/mocks/middleware/resize';
+import harness from '@dojo/framework/testing/harness';
+
+import MyWidget from './MyWidget';
+
+describe('MyWidget', () => {
+	it('test', () => {
+		// create the resize mock
+		const resizeMock = createResizeMock();
+		// pass the resize mock to the harness so it knows to replace the original
+		// middleware
+		const h = harness(() => <App key="app" />, { middleware: [[resize, resizeMock]] });
+
+		// call harness.expect as usual
+		h.expect(() => <div key="root">null</div>);
+
+		// use the resize mock to set the expected return of the resize middleware
+		// by key
+		resizeMock('root', { width: 100 });
+
+		// assert again with the updated expectation
+		h.expect(() => <div key="root">{`{"width":100}`}</div>);
+	});
+});
+```
+
+##### Mock `Store` Middleware
+
+Using `createMockStoreMiddleware` from `@dojo/framework/testing/mocks/middleware/store` creates a typed mock store middleware, which optionally supports mocking processes. To mock a store process pass a tuple of the original store process and the stub process. The middleware will swap out the call to the original process for the passed stub. If no stubs are passed, the middleware will simply no-op all process calls.
```

Comment:
```suggestion
Using `createMockStoreMiddleware` from `@dojo/framework/testing/mocks/middleware/store` creates a typed mock store middleware, which optionally supports the mocking processes. To mock a store process, pass a tuple of the original store process and the stub process. The middleware will swap out the call to the original process for the passed stub. If no stubs are passed, the middleware will simply no-op all process calls.
```

Reply:
@dylans this is intended to be "supports mocking processes" not "supports the mocking processes" - I think it makes sense, as it is referring to actually mock store processes.

Reply:
ok